frontend/catalyst/api_extensions/callbacks.py

```diff
@@ -38,6 +40,64 @@
 from catalyst.utils.types import convert_pytype_to_shaped_array
 
 
+def accelerate(func=None, dev=None):
+    """Seamless integration with jax.jit
+
+    The accelerate function can be used to automatically apply `jax.jit` to
+    the input function `func` and be used to specify the device with the `dev`
+    keyword argument. If `dev` is None, then the default device as determined
+    by JAX is used.
+
+    Can be used in the following modes:
+
+
+    ```py
+    @accelerate
+    def identity(x):
+      return x
+
+    @qml.qjit
+    def qjitted_fn(x):
+      return x
+    ```
+
+    ```
+    @accelerate(dev=jax.devices()[0])
+    def identity(x):
+      return x
+
+    @qml.qjit
+    def qjitted_fn(x):
+      return x
+    ```
+
+    or
+
+    ```
+    @qml.qjit
+    def qjitted_fn(x):
+      return accelerate(lambda x: x)
+    ```
```

Comment:
````suggestion
    """Execute a ``jax.jit`` accelerated function on classical
    accelerators such as GPUs from within a qjit-compiled function.
    
    .. note::
    
        ``catalyst.accelerate`` doses not currently support
        differentiation, and cannot be used inside functions that
        :func:`catalyst.grad` is applied to.
        
    Args:
        func (Callable or PjitFunction): The function to be classically
            accelerated from within the qjit-compiled workflow. This
            function can be already just-in-time compiled with JAX via
            the ``jax.jit`` decorator and a specified device. If not,
            it will be implicitly JIT-compiled, and so must be JIT
            compatible.
        dev (jax.Device): the classical accelerator device the JIT-compiled
            function will run on. Available devices can be retrieved via
            ``jax.devices()``. If not provided, the default value of
            ``jax.devices()[0]`` as determined by JAX will be used.

    .. see-also:: :func:`~.pure_callback`, :func:`.debug.callback`.
    
    **Example**
    
    ```py
    @accelerate(dev=jax.devices("gpu")[0])
    def classical_fn(x):
        return jnp.sin(x) ** 2
    
    @qjit
    def hybrid_fn(x):
        y = classical_fn(jnp.sqrt(x)) # will be executed on a GPU
        return jnp.cos(y)
    ```

    In addition, you can accelerate function that have already been
    ``jax.jit`` decorated:
    
    ```py
    @jax.jit
    def classical_fn(x):
        x = jax.device_put(x, jax.local_devices("gpu")[0])
        return jnp.sin(x) ** 2
    
    @qjit
    def hybrid_fn(x):
        y = classical_fn(x) # will be executed on a GPU
        return jnp.cos(y)
    ```
````

Reply:
@erick-xanadu I'm concerned about my second example above, considering this github post here:

https://github.com/google/jax/issues/6040#issuecomment-797654234

Instead, I'm thinking one of two things:

- We have to _force_ users to specify `accelerate(device)`, or

- We somehow allow the following syntax:

  ```python
      @jax.jit
    def classical_fn(x):
        return jnp.sin(x) ** 2
    
    @qjit
    def hybrid_fn(x):
        x = jax.device_put(x, jax.local_devices("gpu")[0])
        y = classical_fn(x) # will be executed on a GPU
        return jnp.cos(y)
    ```

Would the second option (supporting `jax.device_put` _inside_ `@qjit`) be something we could easily support?

Reply:
When we finalize these examples, we can replace the changelog entry :)

Reply:
I am not sure what the concern is here, with `device_put` and the changes made here https://github.com/PennyLaneAI/catalyst/pull/805/commits/ba899a3473fcafcc4e37b56aea13522a07349db4 all arguments to accelerate will first be placed in function that will be JIT compiled.

Reply:
thanks! My main concern is if a user tries to do 

```python
  @jax.jit
def classical_fn(x):
    return jnp.sin(x) ** 2

@qjit
def hybrid_fn(x):
    x = jax.device_put(x, jax.local_devices("gpu")[0])
    y = classical_fn(x) # will be executed on a GPU
    return jnp.cos(y)
```

But we can consider this out of scope.

Reply:
@erick-xanadu if all good then feel free to commit my docstring changes

Reply:
@josh146 your docstring changes are wrong :sweat_smile:  they are using jax.jit instead of accelerate.

Reply:
I edited them to use accelerate :) You may need to refresh